### PR TITLE
Updated to data stream and pipe method for transmitting data. 

### LINKED ISSUE #300: Implement Pipe Interface
New reports will have an `init` method that has the following signature `function(stream, emitter, callback)`. Pass in the good event emitter, the global read stream, and a callback.

Related to #292

### PR DESCRIPTION
**Pull request by arb**
Closes #300

### CODE REVIEW

File: lib/monitor.js [hunk 1]
```diff
@@ -3,10 +3,12 @@
 var Events = require('events');
 var Http = require('http');
 var Https = require('https');
-var Os = require('os');
 var Hoek = require('hoek');
 var Items = require('items');
+var Os = require('os');
```
Comment (geek) [inline lib/monitor.js]: Os was before Hoek to include node core modules first, then 3rd party, then local.  This needs to be added to the styleguide.

File: lib/monitor.js [hunk 2]
```diff
@@ -3,10 +3,12 @@
...
+var Readable = require('stream').Readable;
```
Comment (geek) [inline lib/monitor.js]: In this case we wouldn't need to create a shortcut like this.  Look at https://github.com/hapijs/hapi/blob/master/lib/response.js#L541

File: lib/monitor.js [hunk 3]
```diff
@@ -115,9 +119,8 @@ internals.Monitor.prototype.start = function (callback) {
         };
     };
 
-    for (var i = 0, il = this.settings.reporters.length; i < il; i++) {
+    Items.serial(this.settings.reporters, function (item, next) {
```
Comment (geek) [inline lib/monitor.js]: why not `for` ?

File: lib/monitor.js [hunk 4]
```diff
@@ -130,26 +133,16 @@ internals.Monitor.prototype.start = function (callback) {
...
 
             Hoek.assert(typeof item.reporter === 'function', 'reporter key must be a constructor function');
+            Hoek.assert(typeof item.events === 'object', 'reporter must specify events to filter on');
 
-            item.args = item.args || [];
-            // this pointer
-            item.args.unshift(null);
-            var constructor = item.reporter.bind.apply(item.reporter, item.args);
-            reporter = new constructor();
+            reporter = new item.reporter(item.events, item.config);
```
Comment (geek) [inline lib/monitor.js]: whats with the lowercase constructor?

File: test/helper.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      if (this.events[data.event]) {
+        this.messages.push(data);
+        this.handler(data);
+      }
+    }.bind(this));
+
+    emitter.once('stop', function () {
+
+      this.stopped = true;
+    }.bind(this));
```
Comment (geek) [inline test/helper.js]: could avoid the bind with a local `self` variable

File: test/monitor.js [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
 
-var Http = require('http');
-var Hapi = require('hapi');
-var Items = require('items');
 var Code = require('code');
-var GoodReporter = require('good-reporter');
+var Hapi = require('hapi');
 var Hoek = require('hoek');
-var Lab = require('lab');
+var Http = require('http');
```
Comment (geek) [inline test/monitor.js]: The original order was better with http first

File: lib/monitor.js (hunk 3 above)
Comment (arb) [inline lib/monitor.js]: Because the reporter `init` function takes a callback.

File: lib/monitor.js (hunk 4 above)
Comment (arb) [inline lib/monitor.js]: It's a configuration option into good, it would be weird to make it upper case.

**Pull request merged**